### PR TITLE
Updates environment variable output

### PR DESCRIPTION
**Pull request by cdrage**
Bolds the info message for outputting the environment variables as well
as changes the grammar.

### CODE REVIEW

**Comment by cdrage** [comment]
Thanks @dharmit :) fixed!

**Comment by cdrage** [comment]
Changed the output a bit, let me know what you think @kadel @girishramnani 

```sh
~/nodejs-ex  master ✗                                                                                                                                                                                                                                                                                                                                              351d ◒  ⍉
▶ odo unlink dh-mysql-apb                                       
 ✓  Service dh-mysql-apb has been successfully unlinked from the component frontend

The below secret environment variables were removed from the 'frontend' component:

· DB_USER
· DB_HOST
· DB_NAME
· DB_PASSWORD
· DB_PORT
· DB_TYPE

~/nodejs-ex  master ✗                                                                                                                                                                                                                                                                                                                                               351d ◒  
▶ odo link dh-mysql-apb  
 ✓  Service dh-mysql-apb has been successfully linked from the component frontend

The below secret environment variables were added to the 'frontend' component:

· DB_PASSWORD
· DB_PORT
· DB_TYPE
· DB_USER
· DB_HOST
· DB_NAME

You can now access the environment variables from within the component pod, for example:
oc rsc <pod-name>
echo $DB_NAME
```

**Comment by dharmit** [comment]
>```
>You can now access the environment variables from within the component pod, for example:
>oc rsc <pod-name>
>echo $DB_NAME
>```

IMHO, this will be confusing to the users who're not familiar with OpenShift terminology. I think we're trying to abstract most things about OpenShift/oc via odo. A user who's not aware of what a `pod` means, would wonder what's going on, no? For a user using odo via VS Code adapters, we will have to choose different message (in json output, I guess) because they're likely not aware of what odo or oc is.

Besides that, `oc rsc <pod-name>` is a command that one has to execute from the development system, right? The statement `You can now access the environment variables from within the component pod, for example` made me think that it's a command that could be executed from inside the pod. But it could be just me.

**Comment by cdrage** [comment]
@dharmit Yeah, I thought the same thing too, but wasn't too sure. I'll remove that output.

When we merge in https://github.com/openshift/odo/pull/2233 we can simply say use `odo describe` to see what environment variables are available.

**Comment by cdrage** [comment]
It will now look like this:

```sh
You can now access the environment variables from within the component pod, for example:
$DB_NAME is now available within component dh-mysql-apb
```

**Comment by kadel** [comment]
> ```shell
> ▶ odo link dh-mysql-apb  
>  ✓  Service dh-mysql-apb has been successfully linked from the component frontend
> 
> The below secret environment variables were added to the 'frontend' component:
> 
> · DB_PASSWORD
> · DB_PORT
> · DB_TYPE
> · DB_USER
> · DB_HOST
> · DB_NAME
> 
> You can now access the environment variables from within the component pod, for example:
> $DB_NAME is now available within component dh-mysql-apb
> ```

The last statement is not true. ` $DB_NAME is now available within component dh-mysql-apb` it should be ` $DB_NAME is now available within component nodejs-something`.
You are linking nodejs component with mysql. So the environment variables are added to the nodejs component not mysql


/hold

File: pkg/odo/cli/component/common_link.go [hunk 1]
```diff
@@ -122,14 +122,35 @@ func (o *commonLinkOptions) run() (err error) {
...
+
+		// Output the environment variables
+		for i := range secret.Data {
+			fmt.Printf("· %v\n", i)
+		}
+
+		// Retrieve the first variable to use as an example.
+		// Have to use a range to access the map
+		var exampleEnv string
+		for i := range secret.Data {
```
Comment (girishramnani) [inline pkg/odo/cli/component/common_link.go]: why use a range? why not just `secret.Data[0]` it? I already see `secret.Data == 0` check which means `Data` array should have atleast one element.

File: pkg/odo/cli/component/common_link.go [hunk 2]
```diff
@@ -122,14 +122,35 @@ func (o *commonLinkOptions) run() (err error) {
...
+		}
 
-	for i := range secret.Data {
-		fmt.Printf("- %v\n", i)
+		// Output what to do next if first linking...
+		// TODO: Retrieve the podName to add to the example?
+		if o.operationName == "link" {
+			log.Infof(`
+You can now access the environment variables from within the component pod, for example:
+$%s is now available as a variable within component %s`, exampleEnv, o.Component())
```
Comment (girishramnani) [inline pkg/odo/cli/component/common_link.go]: @kadel the example might be incorrect, but the code mentions `o.Component()`

**Comment by girishramnani** [comment]
Th

> > ```shell
> > ▶ odo link dh-mysql-apb  
> >  ✓  Service dh-mysql-apb has been successfully linked from the component frontend
> > 
> > The below secret environment variables were added to the 'frontend' component:
> > 
> > · DB_PASSWORD
> > · DB_PORT
> > · DB_TYPE
> > · DB_USER
> > · DB_HOST
> > · DB_NAME
> > 
> > You can now access the environment variables from within the component pod, for example:
> > $DB_NAME is now available within component dh-mysql-apb
> > ```
> 
> The last statement is not true. ` $DB_NAME is now available within component dh-mysql-apb` it should be ` $DB_NAME is now available within component nodejs-something`.
> You are linking nodejs component with mysql. So the environment variables are added to the nodejs component not mysql
> 
> /hold

@cdrage the issue mentioned by tomas seems to be happening because the first argument is taken as component name so you might have to get the component name from the localConfigInfo or a better option might be to make `o.Component()` populate with the correct value.

File: pkg/odo/cli/component/common_link.go (hunk 1 above)
Comment (cdrage) [reply pkg/odo/cli/component/common_link.go]: secret.Data is a map not an array, and that's why you have to use a range to get the first variable.

Comment (girishramnani) [reply pkg/odo/cli/component/common_link.go]: Hehe, felt confusing.

**Comment by cdrage** [comment]
@kadel @girishramnani 

The output was actually incorrect.. some copy-and-paste foo must of messed it up, but here is the correct output:

```sh
▶ odo link mongodb-persistent                                      
 ✓  Service mongodb-persistent has been successfully linked from the component nodejs-nodejs-ex-eapl

The below secret environment variables were added to the 'nodejs-nodejs-ex-eapl' component:

· admin_password
· database_name
· password
· uri
· username

You can now access the environment variables from within the component pod, for example:
$username is now available as a variable within component nodejs-nodejs-ex-eapl
```

**Comment by girishramnani** [comment]
From the code, it seemed that component name was populated from the first arg, I could be wrong

**Comment by cdrage** [comment]
@girishramnani See line: https://github.com/openshift/odo/blob/083bf1852627793927304f24a61cf21b3011f6ca/pkg/odo/cli/component/common_link.go#L118 previously used in the code for `o.Component()`.

Was simply a bad copy and paste from me. The code is correct :)

**Pull request merged**